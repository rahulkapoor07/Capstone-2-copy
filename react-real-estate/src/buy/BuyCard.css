.zoom {
    transition: transform .1s;
    text-align: center;
}

.zoom:hover {
    transform: scale(1.05);
}
