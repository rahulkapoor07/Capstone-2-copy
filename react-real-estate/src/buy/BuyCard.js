import React, { useContext, useEffect, useState } from "react";
import { Link } from "react-router-dom";
import UserContext from "../useContext/UserContext";
import { FcLikePlaceholder, FcLike } from "react-icons/fc";
import "./BuyCard.css";

const BuyCard = ({status,property_id,primary_photo,address,photos,home_details,home_description,
  home_price,street_view,fav_heart}) => {
  
    const {setdetailOpen,singleHome,addHomeInDB,addHomeInUserDB,currentUser, cityStateData, 
      buyHomes,userSavedHomesFunc, setMessage} = useContext(UserContext);

  const singleHomeData = {status,property_id,primary_photo,address,photos,home_details,
    home_description,home_price,street_view,fav_heart};

    const [isColored, setIsColored] = useState(false);

  const handleClick = (e) => {
    setdetailOpen(true);
    singleHome({ ...singleHomeData });
  };

  useEffect(()=>{
    const changeColor = async ()=>{
      if(isColored){
        const homeData = {property_id: singleHomeData.property_id,status_code: singleHomeData.status};
        const homeUserData = {home_property_id: singleHomeData.property_id,user_username: currentUser.username};
        await addHomeInDB({ ...homeData });
        await addHomeInUserDB({ ...homeUserData });
        console.log(cityStateData);
        if(cityStateData){
          await buyHomes({...cityStateData});
        }else{
          await buyHomes({"city" : "yuba city","state_code":"CA"});
        }
        
        await userSavedHomesFunc({"username":currentUser.username});
        setMessage(true);
      }
    }
    changeColor();
  },[isColored]);

  const handleColor = async () => {
    setIsColored((data) => !data);
  };

  return (
    <>
      {singleHomeData ? (
        <div className="zoom mb-5">
          <div className="col">
            <Link to="/buy" onClick={handleClick}>
              <div
                className="card border border-dark"
                style={{ width: "15rem" }}
              >
                {singleHomeData.primary_photo ? (
                  <img
                    className="card-img-top"
                    src={singleHomeData.primary_photo.href}
                    alt="Card cap"
                    style={{ height: "150px" }}
                  />
                ) : (
                  <img
                    className="card-img-top"
                    src="https://st4.depositphotos.com/14953852/22772/v/1600/depositphotos_227725020-stock-illustration-image-available-icon-flat-vector.jpg"
                    alt="Card cap"
                    style={{ height: "150px" }}
                  />
                )}

                <div className="card-body">
                  <h5 className="card-title">{`$${singleHomeData.home_price}`}</h5>
                </div>
                <ul className="list-group list-group-flush">
                  <li className="list-group-item">{`Home ${singleHomeData.status && singleHomeData.status.split("_").join(" ")}`}</li>
                  <li className="list-group-item">{singleHomeData.address}</li>
                </ul>
              </div>
            </Link>
            <Link onClick={handleColor} to="/buy">
              <div>
                {isColored ? (
                  <FcLike
                    size="2rem"
                    style={{ position: "absolute", top: "5px", right: "25px" }}
                  />
                ) : (
                  <FcLikePlaceholder
                    size="2rem"
                    style={{ position: "absolute", top: "5px", right: "25px" }}
                  />
                )}
              </div>
            </Link>
          </div>
        </div>
      ) : (
        <h3>Loading...</h3>
      )}
    </>
  );
};

export default BuyCard;
