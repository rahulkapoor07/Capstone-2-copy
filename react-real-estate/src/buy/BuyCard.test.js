import { render } from '@testing-library/react';
import BuyCard from './BuyCard';
import { MemoryRouter } from 'react-router-dom';


test("render without crashing", ()=>{
  render(<MemoryRouter><BuyCard/></MemoryRouter>);
});

test("snapshot", ()=>{
  const {asFragment} = render(<MemoryRouter><BuyCard/></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});