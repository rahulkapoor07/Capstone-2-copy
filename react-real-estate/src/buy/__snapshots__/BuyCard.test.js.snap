// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="zoom mb-5"
  >
    <div
      class="col"
    >
      <a
        href="/buy"
      >
        <div
          class="card border border-dark"
          style="width: 15rem;"
        >
          <img
            alt="Card cap"
            class="card-img-top"
            src="https://st4.depositphotos.com/14953852/22772/v/1600/depositphotos_227725020-stock-illustration-image-available-icon-flat-vector.jpg"
            style="height: 150px;"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              $undefined
            </h5>
          </div>
          <ul
            class="list-group list-group-flush"
          >
            <li
              class="list-group-item"
            >
              Home undefined
            </li>
            <li
              class="list-group-item"
            />
          </ul>
        </div>
      </a>
      <a
        href="/buy"
      >
        <div>
          <svg
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="2rem"
            stroke="currentColor"
            stroke-width="0"
            style="position: absolute; top: 5px; right: 25px;"
            version="1"
            viewBox="0 0 48 48"
            width="2rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M34,9c-4.2,0-7.9,2.1-10,5.4C21.9,11.1,18.2,9,14,9C7.4,9,2,14.4,2,21c0,11.9,22,24,22,24s22-12,22-24 C46,14.4,40.6,9,34,9z"
              fill="#FFCDD2"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
